### PR TITLE
add a monitor implementation for ray tune, named automlboard.

### PR DESCRIPTION
**Pull request by LionHeartWang**
## What do these changes do?

we add a monitor implementation for ray tune, named ``AutoMLBoard``.

``AutoMLBoard`` can be used as a monitor who collect information about ray tune jobs and
show the trial information to the frontend.

Usage:

```
usage: run.py [-h] [--logdir LOGDIR] [--port PORT] [--db_address DB_ADDRESS]
              [--sleep_interval SLEEP_INTERVAL] [--log_level LOG_LEVEL]

optional arguments:
  -h, --help                        show this help message and exit
  --logdir LOGDIR                   directory of logs about the jobs' status
  --port PORT                       port of the service, 8008 as default
  --db_address DB_ADDRESS           addaress of the database, use a local sqlite3 if not set
  --sleep_interval SLEEP_INTERVAL   time period of polling, 30 seconds as default     
  --log_level LOG_LEVEL             level of the log, "info" as default
```

As default, sqlite will be used as the database backend engine.
To use mysql as the backend engine, run command like this:

```bash
python run.py --db_address mysql://<hostname>:<port>/<db_name>?user=<user>&password=<password> ...
```

### CODE REVIEW

File: python/ray/automlboard/run.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    django.setup()
+    command = ['manage.py', 'migrate', '--run-syncdb']
+    execute_from_command_line(command)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--logdir", type=str, help="directory of logs about the trials' status")
+    parser.add_argument("--port", type=str, help="port of the service", default="8008")
+    parser.add_argument("--db_address", type=str, help="addaress of the database", default=None)
```
Comment (old-bear) [inline python/ray/automlboard/run.py]: --db, default说明下就是本机的sqlite

File: python/ray/automlboard/run.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    parser.add_argument("--sleep_interval", type=int, help="time period of polling", default=30)
```
Comment (old-bear) [inline python/ray/automlboard/run.py]: --reload_interval, default=5

File: python/ray/automlboard/models/models.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    job_id = models.CharField(max_length=50)
+    name = models.CharField(max_length=20)
+    user = models.CharField(max_length=20)
+    start_time = models.CharField(max_length=50)
+    end_time = models.CharField(max_length=50)
+    success_trials = models.BigIntegerField()
+    failed_trials = models.BigIntegerField()
+    running_trials = models.BigIntegerField()
+    total_trials = models.BigIntegerField()
+    best_result = models.TextField()
```
Comment (old-bear) [inline python/ray/automlboard/models/models.py]: 这个字段重复了，有了best_trial可以用sql的relation做的

File: python/ray/automlboard/models/models.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    best_trial_id = models.CharField(max_length=50)
+    current_round = models.BigIntegerField()
```
Comment (old-bear) [inline python/ray/automlboard/models/models.py]: round这个概念开源里我觉得先删了吧，避免他们误解先

File: python/ray/automlboard/models/models.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    progress = models.BigIntegerField()
```
Comment (old-bear) [inline python/ray/automlboard/models/models.py]: progress也是可以计算的吧？

File: python/ray/automlboard/models/models.py [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+        return self.end_time is not None
+
+
+class TrialRecord(models.Model):
+    """
+    Information of a single AutoML trial of the job
+    """
+    trial_id = models.CharField(max_length=50)
+    job_id = models.CharField(max_length=50)
+    user = models.CharField(max_length=20)
```
Comment (old-bear) [inline python/ray/automlboard/models/models.py]: 有job了，user是多余的

File: python/ray/automlboard/models/models.py [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+    trial_type = models.CharField(max_length=20)
```
Comment (old-bear) [inline python/ray/automlboard/models/models.py]: type放到job里吧，一个job的type事固定的

File: python/ray/automlboard/run.py (hunk 1 above)
Comment (LionHeartWang) [reply python/ray/automlboard/run.py]: 好的

File: python/ray/automlboard/models/models.py [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+    trial_status = models.CharField(max_length=20)
+    start_time = models.CharField(max_length=50)
+    end_time = models.CharField(max_length=50)
+    app_url = models.CharField(max_length=256)
+    metrics = models.CharField(max_length=50)
```
Comment (old-bear) [inline python/ray/automlboard/models/models.py]: 这里metrics也没用了吧？

File: python/ray/automlboard/models/models.py (hunk 6 above)
Comment (LionHeartWang) [reply python/ray/automlboard/models/models.py]: 已去除

File: python/ray/automlboard/models/models.py (hunk 4 above)
Comment (LionHeartWang) [reply python/ray/automlboard/models/models.py]: 已去除

File: python/ray/automlboard/models/models.py (hunk 3 above)
Comment (LionHeartWang) [reply python/ray/automlboard/models/models.py]: 已去除

File: python/ray/automlboard/models/models.py (hunk 8 above)
Comment (LionHeartWang) [reply python/ray/automlboard/models/models.py]: 已去除

File: python/ray/automlboard/models/models.py (hunk 7 above)
Comment (LionHeartWang) [reply python/ray/automlboard/models/models.py]: 已修改

File: python/ray/automlboard/run.py (hunk 2 above)
Comment (LionHeartWang) [reply python/ray/automlboard/run.py]: done

File: python/ray/automlboard/README.md [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+```
+usage: run.py [-h] [--logdir LOGDIR] [--port PORT] [--db_address DB_ADDRESS]
+              [--sleep_interval SLEEP_INTERVAL] [--log_level LOG_LEVEL]
+
+optional arguments:
+  -h, --help                        show this help message and exit
+  --logdir LOGDIR                   directory of logs about the jobs' status
+  --port PORT                       port of the service, 8008 as default
+  --db_address DB_ADDRESS           addaress of the database, use a local sqlite3 if not set
```
Comment (old-bear) [inline python/ray/automlboard/README.md]: 这里也改下

**Comment by guoyuhong** [comment]
You need to send a PR of this change to https://github.com/ray-project/ray.

File: python/ray/automlboard/README.md (hunk 9 above)
Comment (LionHeartWang) [reply python/ray/automlboard/README.md]: 好

**Pull request merged**